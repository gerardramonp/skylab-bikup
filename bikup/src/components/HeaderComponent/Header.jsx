import React from 'react';
import { NavLink } from 'react-router-dom';

import './Header.scss';

function Header() {
	const mobileNav = [
		{
			text: 'ExploreMobile',
			url: '/search',
			image:
				'https://trello-attachments.s3.amazonaws.com/5f4cb639a6f5eb1005114de4/5f53d188f7827833bd79ef9b/214746d15b751953f02eb252fb879874/magnifying-glass.png',
		},
		{
			text: 'MyBikesMobile',
			url: '/bikes',
			image:
				'https://trello-attachments.s3.amazonaws.com/5f4cb639a6f5eb1005114de4/5f53d188f7827833bd79ef9b/ca1db79d00b304cb995a7491f0ce71a0/bike.png',
		},
		{
			text: 'ProfileMobile',
			url: '/user',
			image:
				'https://trello-attachments.s3.amazonaws.com/5f4cb639a6f5eb1005114de4/5f53d188f7827833bd79ef9b/e60b6c14d78fa03c5a1ae02b95f9cbca/user.png',
		},
	];

	const desktopNav = [
		{ text: 'My Bikes', url: '/bikes', image: '' },
		{ text: 'Explore', url: '/search', image: '' },
		{ text: 'Profile', url: '/user', image: '' },
	];

	return (
		<header className='header'>
			<div className='header__mobile'>
				<img
					className='logo'
					src='https://cdn.discordapp.com/attachments/692420285143711814/693437226146594876/LogoGerili.png'
					alt='logo'
				/>
				<div className='mobile__nav'>
					{mobileNav.map((link) => {
						return (
							<NavLink
								to={link.url}
								key={link.text}
								activeClassName='active--mobile'
							>
								<img
									src={link.image}
									alt={link.text}
									className='nav__item'
								/>
							</NavLink>
						);
					})}
				</div>
				<img
					className='nav__menu'
					src='https://trello-attachments.s3.amazonaws.com/5f4cb639a6f5eb1005114de4/5f53d188f7827833bd79ef9b/ea9ccf0f7a5fe6c8d066bb89f204f504/open-menu.png'
					alt='menu'
				/>
			</div>

			<div className='header__desktop'>
				<img
					className='logo'
					src='https://cdn.discordapp.com/attachments/692420285143711814/693437226146594876/LogoGerili.png'
					alt='logo'
				/>
				<p className='appName'>bikUP</p>
				<ul className='header__navigation'>
					{desktopNav.map((link) => {
						return (
							<NavLink
								key={link.text}
								className='navigation__item'
								activeClassName='active-desktop'
								to={link.url}
							>
								{link.text}
							</NavLink>
						);
					})}
				</ul>
				<div className='flex-spacer'></div>
				<button className='login__button login__button--header'>
					Log In
				</button>
			</div>
		</header>
	);
}

export default Header;
