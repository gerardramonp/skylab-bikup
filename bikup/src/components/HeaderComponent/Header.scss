$header-bg-color: #222831;
$main-color: #0c9;
$header-separator-color: #393e46;

.header__desktop {
	display: none;
}

.header {
	display: flex;
	flex-direction: row;
	align-items: center;
	position: fixed;
	height: 50px;
	background-color: $header-bg-color;
	bottom: 0;
	width: 100%;
}

.header__mobile {
	display: flex;
	flex-direction: row;
	align-items: center;
	width: 100%;
	padding: 0 24px;
	color: white;
}

.logo {
	height: 35px;
}

.mobile__nav {
	display: flex;
	flex-direction: row;
	justify-content: space-around;
	align-items: center;
	width: 100%;
}

.nav__item {
	height: 25px;
	transition: all 0.2s ease-in-out;
}

.active--mobile {
	transform: scale(1.1);
}

.nav__menu {
	height: 25px;
}

@media only screen and (min-width: 900px) {
	.header__mobile {
		display: none;
	}

	.general-container {
		margin-top: 75px;
	}

	.header {
		display: flex;
		flex-direction: row;
		align-items: center;
		height: 70px;
		position: fixed;
		top: 0;
		margin-bottom: 300px;
		background-color: $header-bg-color;
		color: whitesmoke;
		width: 100%;
	}

	.header__desktop {
		display: flex;
		flex-direction: row;
		align-items: center;
		height: 70px;
		background-color: $header-bg-color;
		color: whitesmoke;
		width: 100%;
	}

	.header__container {
		display: flex;
		flex-direction: row;
		align-items: center;
	}

	.flex-spacer {
		display: flex;
		flex: 1;
	}

	.logo {
		height: 45px;
		margin: 12px;
	}

	.appName {
		font-size: 32px;
		font-weight: bold;
		margin-right: 32px;
	}

	.header__navigation {
		display: flex;
		flex-direction: row;
	}

	.navigation__item {
		display: flex;
		flex-direction: row;
		align-items: center;
		justify-content: center;
		list-style: none;
		text-align: center;
		text-decoration: none;
		color: whitesmoke;
		height: 70px;
		width: 100px;
		font-weight: 500;
		padding: 12px;
		border-left: 1px solid $header-separator-color;
		transition: all 0.2s ease-in-out;
	}

	.active-desktop {
		background-color: $main-color;
		transform: scale(1.08);
		border: unset;
	}

	.login__button--header {
		margin-right: 12px;
		cursor: pointer;
	}
}
