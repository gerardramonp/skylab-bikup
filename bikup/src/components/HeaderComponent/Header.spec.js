import React from 'react';
import renderer from 'react-test-renderer';
import Header from './Header';
import { BrowserRouter as Router } from 'react-router-dom';

describe('Header Component', () => {
	let tree = null;

	beforeEach(() => {
		tree = renderer.create(
			<Router>
				<Header />
			</Router>
		);
	});

	it('should match', () => {
		expect(tree.toJSON()).toMatchSnapshot();
	});
});
