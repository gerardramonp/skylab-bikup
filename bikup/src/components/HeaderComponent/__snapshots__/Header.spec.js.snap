// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header Component should match 1`] = `
<header
  className="header"
>
  <div
    className="header__mobile"
  >
    <img
      alt="logo"
      className="logo"
      src="https://cdn.discordapp.com/attachments/692420285143711814/693437226146594876/LogoGerili.png"
    />
    <div
      className="mobile__nav"
    >
      <a
        aria-current={null}
        href="/search"
        onClick={[Function]}
      >
        <img
          alt="ExploreMobile"
          className="nav__item"
          src="https://trello-attachments.s3.amazonaws.com/5f4cb639a6f5eb1005114de4/5f53d188f7827833bd79ef9b/214746d15b751953f02eb252fb879874/magnifying-glass.png"
        />
      </a>
      <a
        aria-current={null}
        href="/bikes"
        onClick={[Function]}
      >
        <img
          alt="MyBikesMobile"
          className="nav__item"
          src="https://trello-attachments.s3.amazonaws.com/5f4cb639a6f5eb1005114de4/5f53d188f7827833bd79ef9b/ca1db79d00b304cb995a7491f0ce71a0/bike.png"
        />
      </a>
      <a
        aria-current={null}
        href="/user"
        onClick={[Function]}
      >
        <img
          alt="ProfileMobile"
          className="nav__item"
          src="https://trello-attachments.s3.amazonaws.com/5f4cb639a6f5eb1005114de4/5f53d188f7827833bd79ef9b/e60b6c14d78fa03c5a1ae02b95f9cbca/user.png"
        />
      </a>
    </div>
    <img
      alt="menu"
      className="nav__menu"
      src="https://trello-attachments.s3.amazonaws.com/5f4cb639a6f5eb1005114de4/5f53d188f7827833bd79ef9b/ea9ccf0f7a5fe6c8d066bb89f204f504/open-menu.png"
    />
  </div>
  <div
    className="header__desktop"
  >
    <img
      alt="logo"
      className="logo"
      src="https://cdn.discordapp.com/attachments/692420285143711814/693437226146594876/LogoGerili.png"
    />
    <p
      className="appName"
    >
      bikUP
    </p>
    <ul
      className="header__navigation"
    >
      <a
        aria-current={null}
        className="navigation__item"
        href="/bikes"
        onClick={[Function]}
      >
        My Bikes
      </a>
      <a
        aria-current={null}
        className="navigation__item"
        href="/search"
        onClick={[Function]}
      >
        Explore
      </a>
      <a
        aria-current={null}
        className="navigation__item"
        href="/user"
        onClick={[Function]}
      >
        Profile
      </a>
    </ul>
    <div
      className="flex-spacer"
    />
    <button
      className="login__button login__button--header"
    >
      Log In
    </button>
  </div>
</header>
`;
